helix-core/src/main/java/org/apache/helix/HelixPropertyFactory.java

```diff
@@ -47,17 +49,30 @@ public static HelixPropertyFactory getInstance() {
    * Clients may override these values.
    */
   public HelixManagerProperty getHelixManagerProperty(String zkAddress, String clusterName) {
-    ConfigAccessor configAccessor = new ConfigAccessor(zkAddress);
     CloudConfig cloudConfig;
-    // The try-catch logic is for backward compatibility reason only. Even if the cluster is not set
-    // up yet, constructing a new ZKHelixManager should not throw an exception
+    HelixZkClient zkClient = null;
     try {
-      cloudConfig =
-          configAccessor.getCloudConfig(clusterName) == null ? buildEmptyCloudConfig(clusterName)
-              : configAccessor.getCloudConfig(clusterName);
-    } catch (HelixException e) {
-      cloudConfig = buildEmptyCloudConfig(clusterName);
+      // This configAccessor is a dedicated zkclient because HelixManager is single realm
```

Comment:
Let's mention another reason we want a dedicated client instead of shared client here is for the isolation level that we need.